app/src/app/utils/helpers.ts

```diff
@@ -133,16 +134,25 @@ export const getFeaturesIntersectingCounties = (
 
   if (!countyFeatures?.length) return;
   const fips = countyFeatures[0].properties.STATEFP + countyFeatures[0].properties.COUNTYFP;
+  const {mapDocument, shatterIds} = useMapStore.getState();
+  const filterPrefix = mapDocument?.parent_layer.includes("vtd") ? "vtd:" : ""
+  const cachedParentFeatures = parentIdCache.getFilteredIds(`${filterPrefix}${fips}`).map(id => ({
+    id,
+    source: BLOCK_SOURCE_ID,
+    sourceLayer: mapDocument?.parent_layer,
+  }));
+
+  const childFeatures = shatterIds.children.size
+    ? (Array.from(shatterIds.children).map(id => ({
+        id,
+        source: BLOCK_SOURCE_ID,
+        sourceLayer: mapDocument?.child_layer,
+      })) as any)
+    : [];
 
-  const features = map.queryRenderedFeatures(undefined, {
-    layers,
-  });
-
-  return filterFeatures(
-    features,
-    true,
-    [(feature) => Boolean(feature?.id && feature.id.toString().match(/\d{5}/)?.[0] === fips)]
-  );
+  return filterFeatures([...cachedParentFeatures, ...childFeatures], true, [
+    feature => Boolean(feature?.id && feature.id.toString().match(/\d{5}/)?.[0] === fips),
+  ]);
```

Comment:
Random comment but we should probably start testing the FE, esp. would be easy to add unit tests for little utilities like these even if we don't go full FE testing w/ a framework

Reply:
Here's an issue for that #208